.circleci/base_config.yml

```diff
@@ -331,6 +332,59 @@ jobs:
           destination: hotbackup-test
           path: /tmp/test-workdir
 
+  run-cppcheck:
+    docker:
+      - image: arangodb/build-alpine:3.16-gcc11.2-openssl3.1.2
+    resource_class: medium+
+    steps:
+      - run:
+          name: Prepare container
+          command: |
+            mkdir -p /root/workspace/ 
+            mkdir -p /root/project/ 
+      - attach_workspace:
+          at: "/root/workspace/"
+      - checkout-arangodb:
+          with-submodules: true 
+          destination: "/root/project"
+      - checkout-enterprise:
+          destination: "/root/project/enterprise"
+      - run:
+          name: Install cppcheck 2.11
+          command: |
+            cd /root 
+            wget https://github.com/danmar/cppcheck/archive/2.11.tar.gz 
+            tar zxvf 2.11.tar.gz 
+            cd cppcheck-2.11/ 
+            sed -i 's/"2.11"/"2.11.0"/' cmake/versions.cmake  
+            cmake . 
+            make -j4 FILESDIR=/usr install 
+            ln -s /root/workspace/build/ /root/project/
+      - run:
+          name: Install cppcheck junit formator
+          command: |
+            cd /root 
+            wget https://bootstrap.pypa.io/get-pip.py
+            python3 get-pip.py
+            pip install cppcheck-junit
```

Comment:
Why do we install cppcheck every single run (which requires CI time) instead of persisting it within build container?

Also, build containers have to have specific suffix now according to https://github.com/arangodb/arangodb/tree/devel/scripts/docker.

Reply:
Moved to base image in separate pr, waiting with review/merge of this until after that is approved and new images build.